chia/wallet/db_wallet/db_wallet_puzzles.py

```diff
@@ -94,8 +94,10 @@ def create_mirror_puzzle() -> Program:
 MIRROR_PUZZLE_HASH = create_mirror_puzzle().get_tree_hash()
 
 
-def get_mirror_info(parent_puzzle: Program, parent_solution: Program) -> Tuple[bytes32, List[bytes]]:
-    conditions = parent_puzzle.run(parent_solution)
+def get_mirror_info(
+    parent_puzzle: Union[Program, SerializedProgram], parent_solution: Union[Program, SerializedProgram]
+) -> Tuple[bytes32, List[bytes]]:
+    _, conditions = parent_puzzle.run_with_cost(INFINITE_COST, parent_solution)
```

Comment:
are you sure it's OK to pass a `SerializedProgram` to `Program.run()`? Will it deserialize it correctly and turn it into a `SExp`? I'm not sure it does

Reply:
We call this with the same type for puzzle and solution, so I added an assert to signal that intent.

Reply:
Closing thread, taking Arvid's approval as implicit conversation resolution.